pkg/certificate/providers/config.go

```diff
@@ -275,6 +274,81 @@ func (c *MRCProviderGenerator) getCertManagerOSMCertificateManager(mrc *v1alpha2
 	if err != nil {
 		return nil, fmt.Errorf("error instantiating Jetstack cert-manager client: %w", err)
 	}
-
 	return cmClient, nil
 }
+
+// getCertFromKubernetes is a helper function that loads a certificate from a models.Secret
+func getCertFromKubernetes(ns string, secretName string, computeClient compute.Interface) (*certificate.Certificate, error) {
```

Comment:
can this just be a utility method that doesn't accept a compute client [#4930](https://github.com/openservicemesh/osm/issues/4930), and instead extracts the certificate from a models.Secret object?